components/brave_wallet/common/value_conversion_utils.cc

```diff
@@ -112,6 +123,78 @@ mojom::NetworkInfoPtr ValueToEthNetworkInfo(const base::Value& value,
   return chain.Clone();
 }
 
+mojom::NetworkInfoPtr ParseEip3085Payload(const base::Value& value) {
+  mojom::NetworkInfo chain;
+  const base::Value::Dict* params_dict = value.GetIfDict();
+  if (!params_dict)
+    return nullptr;
+
+  const std::string* chain_id = params_dict->FindString("chainId");
+  if (!chain_id) {
+    return nullptr;
+  }
+  chain.chain_id = *chain_id;
+
+  const std::string* chain_name = params_dict->FindString("chainName");
+  if (chain_name) {
+    chain.chain_name = *chain_name;
+  }
```

Comment:
Have you thought about pulling out common codes between ValueToEthNetworkInfo and ParseEip3085Payload to a  function so we don't need to copy them here and below (Line 174-191)?

Reply:
reduced copypaste here https://github.com/brave/brave-core/pull/14104/commits/55c5f85d11113a30791b804f460723aba05beb9f